src/v/redpanda/admin_server.cc

```diff
@@ -1349,6 +1385,56 @@ void admin_server::register_features_routes() {
 
           co_return std::move(res);
       });
+
+    register_route<superuser>(
+      ss::httpd::features_json::put_feature,
+      [this](std::unique_ptr<ss::httpd::request> req)
+        -> ss::future<ss::json::json_return_type> {
+          auto doc = parse_json_body(*req);
+          apply_validator(feature_put_validator, doc);
+
+          auto feature_name = req->param["feature_name"];
+
+          if (!_controller->get_feature_table()
+                 .local()
+                 .resolve_name(feature_name)
+                 .has_value()) {
+              throw ss::httpd::bad_request_exception("Unknown feature name");
+          }
+
+          cluster::feature_update_action action{.feature_name = feature_name};
+          auto& new_state_str = doc["state"];
+          if (new_state_str == "active") {
+              action.action = cluster::feature_update_action::action_t::
+                administrative_activate;
+          } else if (new_state_str == "disabled") {
+              action.action = cluster::feature_update_action::action_t::
+                administrative_deactivate;
+          } else {
+              throw ss::httpd::bad_request_exception("Invalid state");
+          }
+
+          if (need_redirect_to_leader(model::controller_ntp, _metadata_cache)) {
+              throw co_await redirect_to_leader(*req, model::controller_ntp);
+          }
+
+          auto& fm = _controller->get_feature_manager();
+          std::error_code err;
+          if (ss::this_shard_id() == cluster::feature_manager::backend_shard) {
+              err = co_await fm.local().write_action(action);
+          } else {
+              err = co_await fm.invoke_on(
+                cluster::feature_manager::backend_shard,
```

Comment:
when `ss::this_shard_id() == cluster::feature_manager::backend_shard` is true `invoke_on(backend_shard` will invoke the continuation directly so you can unify this condition around `fm.invoke_on`

Reply:
Done